Source/Core/InputCommon/ControlReference/FunctionExpression.cpp

```diff
@@ -636,6 +636,60 @@ class PulseExpression : public FunctionExpression
   mutable Clock::time_point m_release_time = Clock::now();
 };
 
+// usage: ntimes(input, times = 1)
+class NTimesExpression : public FunctionExpression
+{
+private:
+  ArgumentValidation
+  ValidateArguments(const std::vector<std::unique_ptr<Expression>>& args) override
+  {
+    if (args.size() == 1 || args.size() == 2)
+      return ArgumentsAreValid{};
+    else
+      return ExpectedArguments{"input, [times = 1]"};
```

Comment:
you don't need [], that is used for arguments which can be missing but are ignored (like if they have no fallback/default value). `times = 1` is good